src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -12,12 +12,16 @@ import com.google.api.client.googleapis.json.GoogleJsonResponseException
 import com.google.api.client.json.jackson2.JacksonFactory
 import com.google.api.services.dataproc.Dataproc
 import com.google.api.services.pubsub.PubsubScopes
-import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterRequest, ClusterResponse}
+import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterRequest, ClusterResponse, LeoException}
 import org.broadinstitute.dsde.workbench.model.{ErrorReport, WorkbenchExceptionWithErrorReport}
 import org.broadinstitute.dsde.workbench.leonardo.errorReportSource
+import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.GoogleProject
+
+
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future}
 
+case class CallToGoogleApiFailedException(googleProject: GoogleProject, clusterName:String) extends LeoException(s"Call to Google API failed for $googleProject/$clusterName", StatusCodes.NotFound)
```

Comment:
So we map any Google API failure to a 404 status code in Leo? Should we preserve the Google status/error msg in the Leo response?

Reply:
Nooooooo! We should understand the nature of the Google error and handle it appropriately. If Google gives us 404 we should react differently to if they give us 500. We can punt on some of this for now but "google went wrong sorry" isn't a sufficient response to a non-2xx status code.

Reply:
Ok fine. :) See my comment below, I'm confused under what conditions this exception is thrown.